src/main/java/svc/controllers/CourtController.java

```diff
@@ -32,9 +33,10 @@ CourtsDTO GetCourts() {
 	@RequestMapping(method = RequestMethod.GET, value="/{id}")
 	Court GetCourt(@PathVariable("id") Long id) {
 		if (id == null) {
-			LogSystem.LogEvent("Null id passed to controller");
+			LogSystem.LogEvent("non id passed to controller");
+			throw new IncorrectResultSizeDataAccessException("non courtId passed",1);
 		}
-		
-		return courtManager.GetCourtById(id);
+		Court court = courtManager.GetCourtById(id);

```

Comment:
You would want to check for null here and return a NotFoundException (or whatever exception gets caught by the handler and returns 404).